### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by AbbadoLambily** [Netflix/dgs-framework#981]
same environment
same service code

jmeter pressure test
DGS： 400/sec
graphql-spring-boot-starter： 2200/sec
resful api：2200/sec

where is so much performance consumed ?

**Comment by paulbakker** [comment]
That's interesting and quite surprising! We've run perf tests in the past which showed it was pretty much equal to just a RestController, and haven't noticed any significant changes.
Could you share your test setup so that we can reproduce it?

**Comment by AbbadoLambily** [comment]
> That's interesting and quite surprising! We've run perf tests in the past which showed it was pretty much equal to just a RestController, and haven't noticed any significant changes. Could you share your test setup so that we can reproduce it?

It’s a simple database query, the only difference being@PostMapping  and@DgsQuery

**Comment by AbbadoLambily** [comment]
@DgsQuery
    @PostMapping("/query")
    public List<Member> member(Long id,Integer page, Boolean sameCity) {
        return memberService.query(id, page, 20, sameCity);
    }

memberService
public List<Member> query(Long id, Integer pageIndex, Integer pageSize, Boolean sameCity) {
....       
return list;
    }

@paulbakker

**Comment by AbbadoLambily** [comment]
> ```
> @DgsQuery
> @PostMapping("/query")
> public List<Member> member(Long id,Integer page, Boolean sameCity) {
>     return memberService.query(id, page, 20, sameCity);
> }
> ```
> 
> memberService public List query(Long id, Integer pageIndex, Integer pageSize, Boolean sameCity) { .... return list; }
> 
> @paulbakker

DgsQuery   400/sec
PostMapping  2200/sec

One service instance, two request methods, and the performance varies greatly

**Comment by AbbadoLambily** [comment]
After testing, querying a Boolean type field will degrade a lot @paulbakker

**Comment by AbbadoLambily** [comment]
After testing, the returned results have multiple Boolean types or other types, and the performance is much degraded @paulbakker

**Comment by paulbakker** [comment]
Do you mean using scalars? Maybe you can share an example of the result which you think is causing the problem?

**Comment by berngp** [comment]
@AbbadoLambily I would really help if you share with us the project, or at least your GraphQL Schema and GraphQL Queries.

**Comment by AbbadoLambily** [comment]
Test code has been uploaded to my github
Https://github.com/abbadolambily/dgs-demo.git
After the service is running, use Test.jmx to perform JMETER pressure test

@berngp @paulbakker 
😄

**Comment by AbbadoLambily** [comment]
![image](https://user-images.githubusercontent.com/46710794/163519289-3e0826ea-834e-4558-98de-c0c5be9718c5.png)

**Comment by AbbadoLambily** [comment]
@berngp Did you find the reason? 😔

**Comment by kilink** [comment]
When I run your JMeter GraphQL test plan, I get ~1650 requests / sec with DGS v4.9.25.

I also tried it with the latest release candidate (v4.10.0-rc.4), as we made some potential performance improvements there, and I got ~2150 requests / sec, about a 30% increase.

When I run your JMeter test plan against the REST endpoint, I get 17,000 requests / sec. However, I am not sure it's a fair comparison, as your REST endpoint does not take a request body, whereas the GraphQL endpoint obviously does, and has to parse the query on each request, etc.

Other relevant details about my setup:

JVM: 11.0.13-zulu
Heap size: 16gb
GC: G1
